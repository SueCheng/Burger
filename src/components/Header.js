import React, { Component } from "react";
import { Menu, Image, Icon } from "semantic-ui-react";
import Media from "react-media";
import { NavLink } from "react-router-dom";

class Header extends Component {
  render() {
    return (
      <div className="header">
        <Media query="(max-width:650px)">
          {matches =>
            matches ? (
              <Menu color="brown" size="massive" inverted>
                <Menu.Item onClick={this.props.toggleSidebar}>
                  <Icon name="bars" size="big" />
                </Menu.Item>
                <Menu.Menu position="right">
                  <Menu.Item>
                    <div className="imgBackground">
                      <Image
                        size="tiny"
                        src={process.env.PUBLIC_URL + "/img/burger-logo.png"}
                        alt="Burger"
                      />
                    </div>
                  </Menu.Item>
                </Menu.Menu>
              </Menu>
            ) : (
              <Menu color="brown" size="massive" inverted>
                <Menu.Item header>
                  <div className="imgBackground">
                    <Image
                      size="tiny"
                      src={process.env.PUBLIC_URL + "/img/burger-logo.png"}
                      alt="Burger"
                    />
                  </div>
                </Menu.Item>
                <Menu.Menu position="right">
                  <Menu.Item as={NavLink} exact to="/">
                    Burger Builder
                  </Menu.Item>
                  <Menu.Item as={NavLink} exact to="/shoppingcart">
                    <Icon name="cart" />
                    Cart
                  </Menu.Item>
                  {this.props.renderAuthContent()}
                </Menu.Menu>
              </Menu>
            )}
        </Media>
      </div>
    );
  }
}
export default Header;
