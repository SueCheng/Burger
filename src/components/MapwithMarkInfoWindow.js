import React, { Component } from "react";
import { compose } from "recompose";
import {
  withScriptjs,
  withGoogleMap,
  GoogleMap,
  Marker,
  InfoWindow
} from "react-google-maps";
import { Card, Icon } from "semantic-ui-react";

//onCloseClick should go to corresponding page link
class Map extends Component {
  constructor(props) {
    super(props);
    this.state = { isOpen: new Array(props.markNumber).fill(false) };
  }
  onToggleOpen(index) {
    let tmp = this.state.isOpen;
    tmp[index] = !tmp[index];
    this.setState({ isOpen: tmp });
  }
  render() {
    const { zoom, restaurantInfo } = this.props;
    return (
      <GoogleMap
        defaultZoom={zoom}
        defaultCenter={{
          lat: restaurantInfo[0].lat,
          lng: restaurantInfo[0].lng
        }}
      >
        {restaurantInfo.map((restaurant, index) => {
          return (
            <Marker
              key={index}
              position={{ lat: restaurant.lat, lng: restaurant.lng }}
              onClick={() => this.onToggleOpen(index)}
            >
              {this.state.isOpen[index] && (
                <InfoWindow onCloseClick={() => this.onToggleOpen(index)}>
                  <Card onClick={() => this.props.gotoMenu(restaurant.menu)}>
                    <Card.Content header={restaurant.address} />
                    <Card.Content description={`buy ${restaurant.menu}`} />
                    <Card.Content extra>
                      <Icon name="favorite" />
                      4 Reviews
                    </Card.Content>
                  </Card>
                </InfoWindow>
              )}
            </Marker>
          );
        })}
      </GoogleMap>
    );
  }
}

const MapwithMarkInfoWindow = compose(withScriptjs, withGoogleMap)(Map);

export default MapwithMarkInfoWindow;
