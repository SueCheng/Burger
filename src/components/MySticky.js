import React, { Component } from "react";

export default class MySticky extends Component {
  constructor(props) {
    super(props);
    this.myRef = React.createRef();
    this.state = { ticking: false, lastScrollTop: 0 };
    if (this.props.push) this.state = { ticking: false, style: {} };
    else
      this.state = {
        ticking: false,
        style: { position: "fixed", width: "15%", top: props.top }
      };
  }

  componentDidMount() {
    if (this.props.push) {
      this.stickFunc = this.updateStick.bind(this);
      document.body.addEventListener("scroll", this.stickFunc, false);
    }
  }
  componentWillUnmount() {
    if (this.props.push)
      document.body.removeEventListener("scroll", this.stickFunc);
  }
  inRange(top, a, b, span) {
    var result = false;
    if (a > b - span && a < b + span) {
      result = true;
    } else result = false;
    //console.log(`top:${top} result:${result}`);
    return result;
  }
  //when scroll too fast,updateStick can't catch that inRange phase
  updateStick() {
    let scrollDown = false;
    const { top, context } = this.props;
    let rect = this.myRef.current.getBoundingClientRect();
    //update lastScrollTop to decide scroll direction
    if (document.body.scrollTop > this.state.lastScrollTop) {
      //scroll down
      scrollDown = true;
    }
    this.setState({ lastScrollTop: document.body.scrollTop });

    //set to fixed if sticky element's top get to the top when scrolling up
    if (
      scrollDown &&
      Object.keys(this.state.style).length === 0 &&
      this.inRange(true, rect.top, top, 20)
    ) {
      console.log("set state style fixed", rect.top);
      this.setState({ style: { position: "fixed", width: 272, top } });
    }
    //recover to static position when upper context 's bottom get to the top when scrolling down
    rect = context.getBoundingClientRect();
    if (
      !scrollDown &&
      Object.keys(this.state.style).length !== 0 &&
      this.inRange(false, rect.bottom, top, 20)
    ) {
      console.log("set state style to static", rect.bottom);
      this.setState({ style: {} });
    }
    this.setState({ ticking: false });
  }
  onScroll() {
    if (!this.state.ticking) {
      window.requestAnimationFrame(this.updateStick().bind(this));
      this.setState({ ticking: true });
    }
  }
  render() {
    return (
      <div style={this.state.style} ref={this.myRef}>
        {this.props.children}
      </div>
    );
  }
}
