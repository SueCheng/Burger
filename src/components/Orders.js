import React, { Component, Fragment } from "react";
import { connect } from "react-redux";
import { fetchOrders } from "../actions";
import _ from "lodash";
import {
  Container,
  Segment,
  Dimmer,
  Loader,
  Table,
  List,
  Message
} from "semantic-ui-react";
import { itemSchema } from "../config/itemSchema.js";
import errorShowModal from "./hoc/errorShowModal";
import axiosInstance from "../axios-instance";

class Orders extends Component {
  componentDidMount() {
    this.props.fetchOrders();
  }
  headerCtrls() {
    return (
      <Table.Header>
        <Table.Row textAlign="center">
          {itemSchema.map(itemKey => {
            return <Table.HeaderCell key={itemKey}>{itemKey}</Table.HeaderCell>;
          })}
        </Table.Row>
      </Table.Header>
    );
  }
  orderItemCtrls(orderItem, index, bigScreen) {
    return (
      <Table.Row key={index} textAlign={bigScreen ? "center" : "left"}>
        {itemSchema.map(itemKey => {
          if (itemKey === "config")
            return (
              <Table.Cell key={itemKey}>
                {JSON.stringify(orderItem[itemKey])}
              </Table.Cell>
            );
          else
            return <Table.Cell key={itemKey}>{orderItem[itemKey]}</Table.Cell>;
        })}
      </Table.Row>
    );
  }
  orderTable(order, index) {
    let bigScreen = true;
    if (window.matchMedia("(max-width:650px)").matches) bigScreen = false;
    return (
      <Fragment key={index}>
        <Table structured celled>
          {bigScreen && this.headerCtrls()}
          <Table.Body>
            {order.items.map((orderItem, index) =>
              this.orderItemCtrls(orderItem, index, bigScreen)
            )}
          </Table.Body>
        </Table>
        <List divided>
          <List.Item>
            <List.Content floated={bigScreen ? "right" : "left"}>
              Total Price {order.totalPrice.toFixed(1)}
            </List.Content>
          </List.Item>
          <List.Item>
            <List.Content floated={bigScreen ? "right" : "left"}>
              Address {order.address}
            </List.Content>
          </List.Item>
          <List.Item>
            <List.Content floated={bigScreen ? "right" : "left"}>
              Status {order.status}
            </List.Content>
          </List.Item>
        </List>
      </Fragment>
    );
  }
  /*orderItemCtrls(orderItem, index) {
    return (
      <Fragment key={index}>
        <Table.Row key={`${index}0`}>
          <Table.Cell rowSpan={orderItem.cartItems.length}>
            {JSON.stringify(orderItem.address)}
          </Table.Cell>
          <Table.Cell rowSpan={orderItem.cartItems.length}>
            {orderItem.status}
          </Table.Cell>
          <Table.Cell>{JSON.stringify(orderItem.cartItems[0])}</Table.Cell>
        </Table.Row>
        {_.slice(orderItem.cartItems, 1).map((cartItem, i) => {
          return (
            <Table.Row key={i}>
              <Table.Cell>{JSON.stringify(cartItem)}</Table.Cell>
            </Table.Row>
          );
        })}
      </Fragment>
    );
  }*/
  renderOrderTables() {
    if (this.props.orderList.length > 0)
      return this.props.orderList.map((order, index) => {
        return this.orderTable(order, index);
      });
    else return <Message>You haven't order anything</Message>;
  }
  render() {
    if (this.props.loading) {
      //show loading
      return (
        <Segment className="zindexLessHead">
          <Dimmer active inverted>
            <Loader size="large" inverted>
              Loading
            </Loader>
          </Dimmer>
          <div className="orderLoadingBackground" />
        </Segment>
      );
    }
    return <Container>{this.renderOrderTables()}</Container>;
  }
}
function mapStateToProps(state) {
  return {
    loading: state.orders.loading,
    orderList: _.omit(state.orders, "loading").orderList
  };
}

Orders = connect(mapStateToProps, { fetchOrders })(Orders);
Orders = errorShowModal(Orders, axiosInstance);
export default Orders;
