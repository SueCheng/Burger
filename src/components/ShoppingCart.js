import React, { Component } from "react";
import { connect } from "react-redux";
import {
  Container,
  Segment,
  Dimmer,
  Loader,
  Table,
  Button
} from "semantic-ui-react";
import { itemSchema } from "../config/itemSchema.js";
import { fetchShoppingcart } from "../actions";
import _ from "lodash";
import { NavLink } from "react-router-dom";
import errorShowModal from "./hoc/errorShowModal";
import axiosInstance from "../axios-instance";
import Aux from "./hoc/aux";
/*
 category: String,
  quantity: Number,
  price: Number,
  config: { type: Map, of: String }
  */
export class ShoppingCart extends Component {
  componentDidMount() {
    if (!this.props.isCheckoutPage)
      //if it's checkout page,don't fetch again
      this.props.fetchShoppingcart();
  }
  headerCtrls() {
    return (
      <Table.Header>
        <Table.Row textAlign="center">
          {itemSchema.map(itemKey => {
            return <Table.HeaderCell key={itemKey}>{itemKey}</Table.HeaderCell>;
          })}
        </Table.Row>
      </Table.Header>
    );
  }
  cartItemCtrls(cartItem, index, bigScreen) {
    return (
      <Table.Row key={index} textAlign={bigScreen ? "center" : "left"}>
        {itemSchema.map(itemKey => {
          if (itemKey === "config")
            return (
              <Table.Cell key={itemKey}>
                {JSON.stringify(cartItem[itemKey])}
              </Table.Cell>
            );
          else
            return <Table.Cell key={itemKey}>{cartItem[itemKey]}</Table.Cell>;
        })}
      </Table.Row>
    );
  }
  gotoPay() {
    if (!this.props.isCheckoutPage && this.props.cartItems.length > 0)
      return (
        <Button as={Aux(NavLink)} exact to="/checkout">
          Go to Pay
        </Button>
      );
  }
  render() {
    if (this.props.loading)
      //show loading
      return (
        <Segment className="zindexLessHead">
          <Dimmer active inverted>
            <Loader size="medium" inverted>
              Loading
            </Loader>
          </Dimmer>
          <div className="scLoadingBackground" />
        </Segment>
      );
    else {
      let bigScreen = true;
      if (window.matchMedia("(max-width:650px)").matches) bigScreen = false;
      return (
        <Container>
          <Table structured celled>
            {bigScreen && this.headerCtrls()}
            <Table.Body>
              {this.props.cartItems.map((cartItem, index) =>
                this.cartItemCtrls(cartItem, index, bigScreen)
              )}
            </Table.Body>
          </Table>
          {this.gotoPay()}
        </Container>
      );
    }
  }
}
function mapStateToProps(state) {
  return {
    loading: state.shoppingcart.loading,
    cartItems: _.omit(state.shoppingcart, "loading").cartItems
  };
}

let WrappedShoppingCart = connect(mapStateToProps, { fetchShoppingcart })(
  ShoppingCart
);
WrappedShoppingCart = errorShowModal(WrappedShoppingCart, axiosInstance);

export default WrappedShoppingCart;
