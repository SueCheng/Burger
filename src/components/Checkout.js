import React, { Component } from "react";
import { Grid } from "semantic-ui-react";
import ShoppingCart from "./ShoppingCart";
import AddressForm from "./AddressForm/AddressForm";

class Checkout extends Component {
  render() {
    return (
      <div>
        <Grid centered>
          <Grid.Row>
            <Grid.Column computer={6}>
              <ShoppingCart />
            </Grid.Column>
          </Grid.Row>
          <Grid.Row>
            <Grid.Column computer={6}>
              <AddressForm />
            </Grid.Column>
          </Grid.Row>
        </Grid>
      </div>
    );
  }
}
export default Checkout;
