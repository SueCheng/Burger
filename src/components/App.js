import React, { Component } from "react";
import { BrowserRouter, Route, NavLink, Switch } from "react-router-dom";
import { connect } from "react-redux";
import { fetchUser } from "../actions";

import Header from "./Header";
import Checkout from "./Checkout";
import Orders from "./Orders";
import NotFound from "./NotFound";
import SignupForm from "./SignupForm";
import SigninLocalForm from "./SigninLocalForm";
import ShoppingCart from "./ShoppingCart";

import BurgerBuilder from "./BurgerBuilder/BurgerBuilder";
import { Sidebar, Menu, Image, Dropdown, Icon } from "semantic-ui-react";
import Media from "react-media";

class App extends Component {
  constructor(props) {
    super(props);
    this.state = { sidebarVisible: false };
  }

  //get auth user
  componentDidMount() {
    this.props.fetchUser();
  }

  componentWillUpdate(nextProps, nextState) {
    if (nextState.sidebarVisible === true) {
      document.addEventListener("keydown", this.handleKeyPress);
    } else {
      document.removeEventListener("keydown", this.handleKeyPress);
    }
  }
  handleKeyPress = e => {
    if (e.keyCode === 27 && this.state.sidebarVisible) {
      this.setState({ sidebarVisible: false });
    }
  };

  hideSidebarIfVisible() {
    if (this.state.sidebarVisible) {
      this.setState({
        sidebarVisible: false
      });
    }
  }
  toggleSidebar() {
    this.setState({
      sidebarVisible: !this.state.sidebarVisible
    });
  }
  shortenLongString(str) {
    if (str.length > 10) {
      return str.substr(0, 10) + "...";
    }
  }
  renderUserNameId(auth) {
    if (auth) {
      switch (auth.loginMethod) {
        case "facebook":
          return `Welcome ${this.shortenLongString(auth.facebook.facebookId)}`;
        case "google":
          return `Welcome ${this.shortenLongString(auth.google.googleId)}`;
        case "local":
          return `Welcome ${this.shortenLongString(auth.local.userName)}`;
        default:
          return null;
      }
    }
    return null;
  }
  renderAuthContent() {
    if (this.props.auth)
      return (
        <Dropdown item text={this.renderUserNameId(this.props.auth)}>
          <Dropdown.Menu>
            <Dropdown.Item
              as={NavLink}
              onClick={(event, data) => {
                this.setState({ sidebarVisible: false });
              }}
              exact
              to="/orders"
              text="Orders"
            />
            <Dropdown.Item as="a" href="/auth/logout" text="Logout" />
          </Dropdown.Menu>
        </Dropdown>
      );
    else {
      return (
        <Dropdown item text="Sign up or Login">
          <Dropdown.Menu>
            <Dropdown.Item
              as="a"
              icon="facebook"
              href="/auth/facebook"
              text="Login with Facebook"
            />
            <Dropdown.Item
              as="a"
              icon="google"
              href="/auth/google"
              text="Login with Google"
            />
            <Dropdown.Item
              as={NavLink}
              exact
              to="/signinlocal"
              icon="user"
              text="Login Locally"
              onClick={(event, data) => {
                this.setState({ sidebarVisible: false });
              }}
            />
            <Dropdown.Item
              as={NavLink}
              exact
              to="/signup"
              icon="signup"
              text="Signup"
              onClick={(event, data) => {
                this.setState({ sidebarVisible: false });
              }}
            />
          </Dropdown.Menu>
        </Dropdown>
      );
    }
  }
  render() {
    return (
      <BrowserRouter>
        <div>
          <Sidebar.Pushable>
            <Media
              query="(max-width:650)" //650px could insure the right menu showed completely
              render={() => (
                <Sidebar
                  as={Menu}
                  animation="overlay"
                  width="wide"
                  visible={this.state.sidebarVisible}
                  vertical
                >
                  <Menu.Item header>
                    <div className="imgBackground">
                      <Image
                        size="tiny"
                        src={process.env.PUBLIC_URL + "/img/burger-logo.png"}
                        alt="Burger"
                      />
                    </div>
                  </Menu.Item>
                  <Menu.Item
                    as={NavLink}
                    exact
                    to="/"
                    onClick={(event, data) => {
                      this.setState({ sidebarVisible: false });
                    }}
                  >
                    Burger Builder
                  </Menu.Item>
                  <Menu.Item
                    as={NavLink}
                    exact
                    to="/shoppingcart"
                    onClick={(event, data) => {
                      this.setState({ sidebarVisible: false });
                    }}
                  >
                    <Icon name="cart" />
                    Cart
                  </Menu.Item>
                  {this.renderAuthContent()}
                </Sidebar>
              )}
            />
            <Sidebar.Pusher onClick={this.hideSidebarIfVisible.bind(this)}>
              <Header
                toggleSidebar={this.toggleSidebar.bind(this)}
                renderAuthContent={this.renderAuthContent.bind(this)}
              />
              <Switch>
                <Route exact path="/signup" component={SignupForm} />
                <Route exact path="/signinlocal" component={SigninLocalForm} />
                <Route exact path="/" component={BurgerBuilder} />
                <Route
                  exact
                  path="/shoppingcart"
                  render={props => (
                    <ShoppingCart isCheckoutPage={false} {...props} />
                  )}
                />
                <Route exact path="/orders" component={Orders} />
                <Route exact path="/checkout" component={Checkout} />
                <Route component={NotFound} />
              </Switch>
            </Sidebar.Pusher>
          </Sidebar.Pushable>
        </div>
      </BrowserRouter>
    );
  }
}

function mapStateToProps({ auth }) {
  return { auth };
}
App = connect(mapStateToProps, { fetchUser })(App);
export default App;
