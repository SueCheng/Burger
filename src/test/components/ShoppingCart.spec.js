import { ShoppingCart } from "../../components/ShoppingCart";
import { shallow } from "enzyme";
import React from "react";

describe("create snapshot for givin fake data", () => {
  let tree;
  beforeEach(() => {
    tree = shallow(
      <ShoppingCart fetchShoppingcart={jest.fn()} loading={true} />
    );
  });
  test(" snapshot with loading", () => {
    expect(tree).toMatchSnapshot();
  });

  test(" snapshot with no cartItem and loading false", () => {
    tree.setProps({
      loading: false,
      cartItems: []
    });
    expect(tree).toMatchSnapshot();
  });
  test(" snapshot with 2 cartItems", () => {
    tree.setProps({
      loading: false,
      cartItems: [
        {
          category: "burger",
          quantity: 1,
          price: 3.2,
          config: { salad: 0, cheese: 1, bacon: 0, meat: 0 }
        },
        {
          category: "burger",
          quantity: 1,
          price: 14.5,
          config: { salad: 0, cheese: 1, bacon: 2, meat: 1 }
        }
      ]
    });
    expect(tree).toMatchSnapshot();
  });
});
