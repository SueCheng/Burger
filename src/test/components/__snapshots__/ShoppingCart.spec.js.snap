// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot for givin fake data  snapshot with 2 cartItems 1`] = `
<Container>
  <Table
    as="table"
    celled={true}
    structured={true}
  >
    <TableHeader
      as="thead"
    >
      <TableRow
        as="tr"
        cellAs="td"
        textAlign="center"
      >
        <TableHeaderCell
          as="th"
          key="category"
        >
          category
        </TableHeaderCell>
        <TableHeaderCell
          as="th"
          key="quantity"
        >
          quantity
        </TableHeaderCell>
        <TableHeaderCell
          as="th"
          key="price"
        >
          price
        </TableHeaderCell>
        <TableHeaderCell
          as="th"
          key="config"
        >
          config
        </TableHeaderCell>
      </TableRow>
    </TableHeader>
    <TableBody
      as="tbody"
    >
      <TableRow
        as="tr"
        cellAs="td"
        key="0"
        textAlign="center"
      >
        <TableCell
          as="td"
          key="category"
        >
          burger
        </TableCell>
        <TableCell
          as="td"
          key="quantity"
        >
          1
        </TableCell>
        <TableCell
          as="td"
          key="price"
        >
          3.2
        </TableCell>
        <TableCell
          as="td"
          key="config"
        >
          {"salad":0,"cheese":1,"bacon":0,"meat":0}
        </TableCell>
      </TableRow>
      <TableRow
        as="tr"
        cellAs="td"
        key="1"
        textAlign="center"
      >
        <TableCell
          as="td"
          key="category"
        >
          burger
        </TableCell>
        <TableCell
          as="td"
          key="quantity"
        >
          1
        </TableCell>
        <TableCell
          as="td"
          key="price"
        >
          14.5
        </TableCell>
        <TableCell
          as="td"
          key="config"
        >
          {"salad":0,"cheese":1,"bacon":2,"meat":1}
        </TableCell>
      </TableRow>
    </TableBody>
  </Table>
  <Button
    as={[Function]}
    exact={true}
    to="/checkout"
  >
    Go to Pay
  </Button>
</Container>
`;

exports[`create snapshot for givin fake data  snapshot with loading 1`] = `
<Segment>
  <Dimmer
    active={true}
    inverted={true}
  >
    <Loader
      inverted={true}
      size="medium"
    >
      Loading
    </Loader>
  </Dimmer>
  <div
    className="scLoadingBackground"
  />
</Segment>
`;

exports[`create snapshot for givin fake data  snapshot with no cartItem and loading false 1`] = `
<Container>
  <Table
    as="table"
    celled={true}
    structured={true}
  >
    <TableHeader
      as="thead"
    >
      <TableRow
        as="tr"
        cellAs="td"
        textAlign="center"
      >
        <TableHeaderCell
          as="th"
          key="category"
        >
          category
        </TableHeaderCell>
        <TableHeaderCell
          as="th"
          key="quantity"
        >
          quantity
        </TableHeaderCell>
        <TableHeaderCell
          as="th"
          key="price"
        >
          price
        </TableHeaderCell>
        <TableHeaderCell
          as="th"
          key="config"
        >
          config
        </TableHeaderCell>
      </TableRow>
    </TableHeader>
    <TableBody
      as="tbody"
    />
  </Table>
</Container>
`;
